NguyenThuPhuong/phuong-app/app/Http/Controllers/HomeController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Models\Products;
+use App\Models\Product;
+
+class HomeController extends Controller
+{
+    public function index()
+    {
+        $products = DB::table('products')->paginate(8);
+
+        return view('home-page', ['products' => $products]);
+    }
+    public function shop()
```

Comment:
```suggestion

    public function shop()
```
Trước khi mở khối thì nên có 1 line trắng